pyro/distributions/conditional.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+
+import torch
+import torch.nn
+
+
+class ConditionalDistribution(object):
+    @abstractmethod
+    def condition(self, context):
+        """:rtype: torch.distributions.Distribution"""
+        raise NotImplementedError
+
+
+class ConditionalTransform(object):
+    @abstractmethod
+    def condition(self, context):
+        """:rtype: torch.distributions.Transform"""
+        raise NotImplementedError
+
+
+class ConditionalTransformModule(ConditionalTransform, torch.nn.Module):
+    """
+    Conditional transforms with learnable parameters such as normalizing flows should inherit from this class rather
+    than `ConditionalTransform` so they are also a subclass of `nn.Module` and inherit all the useful methods of that
```

Comment:
nit: add a sphinx link via
```
than :class:`ConditionalTransform` so they are ...
```

Reply:
I believe you need to fully qualify references to classes in different modules, but you can tell sphinx to hide the qualification with a tilde:
```
blah blah :class:`~pyro.distributions.conditional.ConditionalTransform` blah blah
```

Reply:
Okay, made that change!